extensions/chat/client.mjs

```diff
@@ -40,33 +40,103 @@ class Client {
 
   send_msg(client){
     return async function(evt){
-      if(evt.keyCode == 13){
-        let message_to_id = document.getElementById('message_to');
-        let chat_message_id = document.getElementById('chat_message');
-        if(message_to_id.value){
+      if(evt.keyCode === 13){
+        const message_to_id = document.getElementById('message_to');
+        const chat_message_id = document.getElementById('chat_message');
+        if(chat_message_id.value.substring(0, 2) === "!/"){
+            client.handleCommand(chat_message_id.value.trim());
+        }
+        else if(message_to_id.value){
           client.helper.callC2sAPI(null, 'privateMessage', 5000, {'msg_to': message_to_id.value, 'msg': chat_message_id.value});
-          chat_message_id.value = '';
         }
         else{
           client.helper.callC2sAPI(null, 'broadcastMessage', 5000, {'msg': chat_message_id.value});
-          chat_message_id.value = '';
         }
+        chat_message_id.value = '';
+      }
+    }
+  }
+    
+  /**
+   * List the user command for player
+   * @listCommand
+   */
+  listCommand(){
+    document.getElementById('message_history').innerHTML += "<pre>" + 
+    this.HTMLEncode(`
+Usage: !/<Command> <arg1> <arg2> ...
+  !/teleport <Map> <x> <y>    Teleport to the map Map at coordinate (x, y)
+  !/help                List command usages
+    `) + "</pre>";
+    
+  }
+
+  /**
+   * Handle the command from client
+   * @handleCommand
+   * @param {string} cmd - the command including arguments
+   */
+  async handleCommand(cmd){
+    if(cmd === '!/help'){
+      this.listCommand();
+    }
+    else if(cmd.split(" ")[0] === '!/teleport'){
+      var mapCoord = this.helper.gameClient.playerInfo.mapCoord;
+      mapCoord.mapName = cmd.split(" ")[1];
+      mapCoord.x = cmd.split(" ")[2];
+      mapCoord.y = cmd.split(" ")[3];
+      if(isNaN(parseInt(mapCoord.x, 10)) || isNaN(parseInt(mapCoord.y, 10))){
+        document.getElementById('message_history').innerHTML += '<span>Invalid Coordinate</span><br>';
+        this.listCommand();
+      }
+      else{
+        this.helper.callC2sAPI(null, 'teleport', 5000, mapCoord);
+      }
+    }
+    else{
+      const result = await Promise.resolve(this.helper.callC2sAPI(null, 'adminCommand', 5000));
```

Comment:
[Optional] Maybe call it otherCommands or hiddenCommands? We can have hidden commands for escape room as well.

Reply:
Done. Change to otherCommands